packages/aws-cdk-lib/aws-ec2/test/cfn-init.test.ts

```diff
@@ -136,6 +136,60 @@ test('empty configs are not rendered', () => {
   });
 });
 
+test('duplicate config arguments not deduplicated', () => {
+  //GIVEN
+  const config = new ec2.InitConfig([
+    ec2.InitCommand.argvCommand([
+      'useradd', '-u', '1001', '-g', '1001', 'eguser',
+    ]),
+    ec2.InitCommand.argvCommand([
+      'useradd', '-a', '-u', '1001', '-g', '1001', 'eguser',
+    ]),
+  ]);
+
+  // WHEN
+  const init = ec2.CloudFormationInit.fromConfigSets({
+    configSets: { default: ['config'] },
+    configs: { config },
+  });
+  init.attach(resource, linuxOptions());
+
+  // THEN
+  expectMetadataLike({
+    'AWS::CloudFormation::Init': {
+      configSets: {
+        default: ['config'],
+      },
+      config: {
+        commands: {
+          '000': {
+            command: ['useradd', '-u', '1001', '-g', '1001', 'eguser'],
+          },
+          '001': {
+            command: ['useradd', '-a', '-u', '1001', '-g', '1001', 'eguser'],
+          },
+        },
+      },
+    },
+  });
+});
+
+test('deepMerge properly deduplicates non-command arguments', () => {
+  // WHEN
+  const config = new ec2.InitConfig([
+    ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/foo.zip'),
+    ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/foo.zip'),
+    ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/foo.zip'),
```

Comment:
can you make this:

```
ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/foo.zip'),
ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/foo.zip'),
ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/bar.zip'),
ec2.InitSource.fromUrl('/tmp/foo', 'https://amazon.com/bar.zip'),
ec2.InitSource.fromUrl('/tmp/bar', 'https://amazon.com/bar.zip'),
ec2.InitSource.fromUrl('/tmp/bar', 'https://amazon.com/bar.zip'),
ec2.InitSource.fromUrl('/tmp/bar', 'https://amazon.com/foobar.zip'),
ec2.InitSource.fromUrl('/tmp/bar', 'https://amazon.com/foobar.zip'),
```

or something like this so we can see whether the dedeup logic applies to both the first parameter and the second, or if it's just the second?

Reply:
Good catch, will include some more dedup test values.